applications/FluidDynamicsApplication/python_scripts/adaptative_remeshing_fluid_dynamics_analysis.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import print_function, absolute_import, division  # makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+
+# Importing Kratos
+import KratosMultiphysics
+
+# Other imports
+import sys
+
+# Import the base structural analysis
+from KratosMultiphysics.FluidDynamicsApplication.fluid_dynamics_analysis import FluidDynamicsAnalysis as BaseClass
+
+class AdaptativeRemeshingFluidDynamicsAnalysis(BaseClass):
+    """
+    This class is the main-script of the FluidDynamicsApplication when using adaptative remeshing put in a class
+
+    It can be imported and used as "black-box"
+    """
+
+    def RunSolutionLoop(self):
+        """This function executes the solution loop of the AnalysisStage
+        It can be overridden by derived classes
+        """
+
+        # If we remesh using a process
+        computing_model_part = self._GetSolver().GetComputingModelPart()
+        root_model_part = computing_model_part.GetRootModelPart()
+
+        while self.time < self.end_time:
+            self.time = self._GetSolver().AdvanceInTime(self.time)
+            # We reinitialize if remeshed previously
+            if root_model_part.Is(KratosMultiphysics.MODIFIED):
+                self._ReInitializeSolver()
+            self.InitializeSolutionStep()
+            # We reinitialize if remeshed on the InitializeSolutionStep
+            if root_model_part.Is(KratosMultiphysics.MODIFIED):
+                self._ReInitializeSolver()
+                self.InitializeSolutionStep()
+            self._GetSolver().Predict()
+            self._GetSolver().SolveSolutionStep()
+            self.FinalizeSolutionStep()
+            self.OutputSolutionStep()
+
+
+    def _CreateProcesses(self, parameter_name, initialization_order):
+        """Create a list of Processes
+        This method is TEMPORARY to not break existing code
+        It will be removed in the future
+        """
+        list_of_processes = super(AdaptativeRemeshingFluidDynamicsAnalysis, self)._CreateProcesses(parameter_name, initialization_order)
+
+        if parameter_name == "processes":
+            processes_block_names = ["recursive_remeshing_process"]
```

Comment:
Why is it always recursive? I'd call it `mesh_adaptivity_procresses` for the sake of generality.